### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by amal-viswa** [requarks/wiki#5565] [Help / Questions]
Hi ,
 I'm unable use any of the authentication modules listed in wiki.js, while i'm able to configure the modules as mentioned in the  " https://docs.requarks.io/auth " but at the end i couldn't able to login using the external idp's.

-  i have tried **Azure AD, Saml, slack, and google**, i'm getting the same error for all shown below

for azure 

![Screenshot 2022-08-23 at 18-12-04 Error](https://user-images.githubusercontent.com/90927394/186163083-0ddd5660-2ab6-4b23-8967-ae9b9f13d517.png)

for slack 

![Screenshot 2022-08-23 at 18-12-26 Error](https://user-images.githubusercontent.com/90927394/186163166-e291b48d-7ba7-4b5e-9f6a-c12ae47c9c11.png)

I've followed the exact steps mentioned in the official document .
 
looking to hear from your end regarding this issue

**Comment by xMetro** [comment]
Did you set your Cookie Encryption Key String. It something with Chrome Cookie.

I got AAD working with 44 letters only

**Comment by jidckii** [comment]
![image](https://user-images.githubusercontent.com/6608896/211534285-0e4b6721-6b16-4136-8425-779b4f8061e7.png)
I have the same problem, set up google, but 500 is always returned, the server logs are empty about this

**Comment by jidckii** [comment]
Gitlab same

**Comment by jidckii** [comment]
Well, it certainly is very strange!
Users are not automatically mapped, they must be manually created for the desired provider with the correct email, then the login works.
Otherwise, it's just a 500 error and that's it, in the server log in debug mode there is not a word about it at all...
I just happened to try it.
![image](https://user-images.githubusercontent.com/6608896/211574473-550278f8-8fc5-4c1c-a82a-f57cc4978517.png)
@amal-viswa  FYI

Is this by design, or is it some kind of bug?

```
Current Version
2.5.295

PostgreSQL
15.1
```

**Reply by NGPixel** [reply]
You need to enable Self-registration [as detailed in the docs](https://docs.requarks.io/auth#self-registration).

**Reply by SimonCoutureCSTJ** [reply]
@NGPixel  I'm using azure on my side with self-registration. But I still get the error. Any idea?
